### PR TITLE
Two improvements to API for BFAST feature

### PR DESCRIPTION
**Pull request by oskooi**
Addresses two of the three recommendations in [#2609 (comment)](https://github.com/NanoComp/meep/pull/2609#issuecomment-1826102961) for improving the API of the recently added BFAST feature:
1. Eliminate the boolean parameter `need_bfast`.
2. Replace `bfast_kbar` with `bfast_scaled_k` which can be one of two types: `Vector3` or `None`.

The third recommendation for setting the Courant factor automatically will be added in a separate PR (pending further tests).

### CODE REVIEW

File: src/step_db.cpp [hunk 1]
```diff
@@ -62,6 +62,7 @@ bool fields_chunk::step_db(field_type ft) {
         realnum *f_p = have_p ? f[c_p][cmp] : NULL;
         realnum *f_m = have_m ? f[c_m][cmp] : NULL;
         realnum *the_f = f[cc][cmp];
+        bool need_bfast = (bfast_scaled_k[0] || bfast_scaled_k[1] || bfast_scaled_k[2]) ? 1 : 0;
```
Comment (stevengj) [inline src/step_db.cpp:65]: The result of `||` should already be a boolean value (`1` or `0`).
```suggestion
        bool need_bfast = bfast_scaled_k[0] || bfast_scaled_k[1] || bfast_scaled_k[2];
```

File: python/tests/test_refl_angular.py [hunk 2]
```diff
@@ -47,13 +47,13 @@ def reflectance_angular(
         theta_rad = math.radians(theta_deg)
 
         if need_bfast:
```
[the remark this answers is not in the document]
Comment (stevengj) [reply python/tests/test_refl_angular.py:49]: oh, I see, it is the function argument.

[the remark this answers is not in the document]
Comment (oskooi) [reply python/tests/test_refl_angular.py]: Done.

File: src/step_db.cpp (hunk 1 above)
Comment (oskooi) [reply src/step_db.cpp]: Fixed.

**Pull request merged**